### PR TITLE
NurseConnect change management

### PR DESCRIPTION
**Pull request by gsvr**

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: nursereg/tasks.py [hunk 1]
```diff
@@ -317,7 +334,27 @@ def update_create_vumi_contact(nursereg_id, client=None, sender=None):
             except:
                 logger.error('Problem contacting http_api', exc_info=True)
 
-            return updated_contact
+            cm_active_nr_subs = Subscription.objects.filter(
+                to_addr=nursereg.cmsisdn, active=True,
+                message_set__short_name="nurseconnect")
```
Comment (imsickofmaps) [inline nursereg/tasks.py:341]: This is only going to work as long as there's only one nurse related messageset. I don't think we need to address this now, but please note so in a comment

File: nursereg/tasks.py [hunk 2]
```diff
@@ -255,6 +259,32 @@ def create_subscription(contact, sender=None):
...
+        subscription.save()
+
+        # deactivate active subscriptions for rmsisdn
+        sub.active = False
+        sub.save()
+
+    # TODO #123: Clear extras for old contact for external change requests
+
+    # return the last created subscription
+    cm_active_nr_subs = Subscription.objects.filter(
```
Comment (imsickofmaps) [inline nursereg/tasks.py]: This is only going to work as long as there's only one nurse related messageset. I don't think we need to address this now, but please note so in a comment

File: nursereg/tasks.py [hunk 3]
```diff
@@ -317,7 +334,27 @@ def update_create_vumi_contact(nursereg_id, client=None, sender=None):
             except:
                 logger.error('Problem contacting http_api', exc_info=True)
 
-            return updated_contact
+            cm_active_nr_subs = Subscription.objects.filter(
```
Comment (imsickofmaps) [inline nursereg/tasks.py]: `cm_active_nr_subs` is a nasty name for a variable :grimacing:

File: nursereg/tests.py [hunk 4]
```diff
@@ -766,31 +830,277 @@ def test_create_vumi_contact_passport(self):
...
+            "nc_id_type": "sa_id",
+            "nc_faccode": "123456",
+            "nc_source_name": "Test Nurse Source",
+            "nc_subscription_type": "11",
+            "nc_subscription_rate": "4",
+            "nc_subscription_seq_start": "13",
+        })
+        postsubs = Subscription.objects.all().count()
+        # check two additional subscriptions created
+        self.assertEqual(postsubs, presubs+2)
```
Comment (imsickofmaps) [inline nursereg/tests.py:1055]: Surely there should only be one additional created? No-one should have two active nurse messagesets?

**Comment by imsickofmaps** [comment]
Minor clarifications and requests for better names (or comments) but generally :+1:

**Pull request merged**